### PR TITLE
Fixed #3899 - Handle new OSX version correctly; reformatted platform-…

### LINKED ISSUE #3899: Nightly CI passes failing with "could not infer target OS" errors from ReportBinarySizes.
***Nightly builds are failing with:***
```
2022-08-24T18:23:10.9888060Z [command]/usr/local/bin/pwsh -NoLogo -NoProfile -NonInteractive -Command . '/Users/runner/work/_temp/988eecad-ed75-4795-ae69-0713e9872959.ps1'
2022-08-24T18:23:11.7682110Z ##[error]Could not infer target OS
2022-08-24T18:23:11.7699480Z ##[error]Could not infer toolchain
2022-08-24T18:23:11.7710610Z ##[error]Could not infer target platform
```

**Additional context**
THe root cause is that our OS image name changed from macOS-10.5 to macos-11.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

### PR DESCRIPTION
**Pull request by LarryOsterman**
…matrix JSON files

Fixed #3899 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by RickWinter — COMMENTED** [review]
Looks fine to me.  Does this need to go in azure-sdk-tools to be replicated everywhere?  I recall some EngSys rule like that for files in eng/....

File: eng/scripts/Get-BinarySizes.ps1 [hunk 1]
```diff
@@ -41,13 +41,16 @@ function setEnvVar($key, $value) {
 }
 
 function getTargetOs {
-    if ($OsVMImage.StartsWith('macOS')) {
+    if ($OsVMImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
```
Comment (benbp) [inline eng/scripts/Get-BinarySizes.ps1:44]: This is fine but FYI here is another powershell mechanism I find to be cleaner:

```
if ($OsVMImage -like "macos*") {
```

**Review by benbp — APPROVED** [review]

**Comment by benbp** [comment]
> Looks fine to me. Does this need to go in azure-sdk-tools to be replicated everywhere? I recall some EngSys rule like that for files in eng/....

@RickWinter this is a C++ specific script, so it does not need to be replicated. We only enforce replication rules for `eng/common/...` paths, and everything else in `eng/` is for language specific configs and scripts.

File: eng/scripts/Get-BinarySizes.ps1 (hunk 1 above)
Comment (LarryOsterman) [reply eng/scripts/Get-BinarySizes.ps1:44]: Oooh, that's way better.

Comment (LarryOsterman) [reply eng/scripts/Get-BinarySizes.ps1:44]: I'll update the script in a future pull request.

**Pull request merged**